docs/specs/src/hwp_3_correctness.md

```diff
@@ -0,0 +1,404 @@
+  block progress; the amount of work that can be done scales
+  linearly with the amount of users (as opposed to blockchain)
+  global consensus mechanisms are essentially single-threaded
+  processes where all users have to advance state simultaneously.
+  \[todo: ACB\]
+
+3.  **Shared-state Finality:** Many blockchains approximate chain
+  finality by assuming that the "longest-chain wins." That strategy
+  does not translate well to agent-centric chains which are simply
+  histories of an agent's actions. While there is no concern about
+  forking global state because a Holochain app doesn't have one, we
+  can imagine a situation where Alice and Bob have countersigned a
+  transaction, and then Alice forks her source chain by later
+  committing an entry to an earlier sequence position in her chain.
+  If she also set back her clock to make it look like Bob
+  participated in a fork instead of her, this could break the
+  cross-chain atomic change he was relying on. This can even happen
+  non-maliciously when someone suffers data loss and restores from a
+  backup after having made changes that did not get included in the
+  backup. While the initial beta version of Holochain does not offer
+  fork finality protections for source chains, later versions will
+  incorporate "meta-data hardening" which enables gossipping peers
+  to tentatively solidify a state of affairs when they see gossip
+  for a time window has calmed and neighbors have converged on the
+  same state. After this settling period (which might get set to
+  something between 5 to 15 minutes) any later changes which would
+  produce a conflict (such as forking a chain) can be rejected.
+
+Security
+--------
+
+The system's resilience to intentional gaming, and disruption by
+malicious actors will be covered in depth in future papers, but here we
+provide an overview.
+
+There are many contributing factors to what allows systems to live up to
+the varying safety and security requirements of their users. In general,
+the approach taken in Holochain is to provide affordances that take into
+account the many types of real-world costs that result from adding
+security and safety to systems such that application developers can
+match the trade-offs of those costs to their application context. The
+integrity guarantees listed in the formal system description, detail the
+fundamental data safety that Holochain applications provide. Some other
+important facets of system security and safety come from:
+
+1.  Gating access functions that change state, for which Holochain
+  provides a unified and flexible Object Capabilities model
+
+2.  Detecting and Blocking participation of bad actors, for which
+  Holochain provides the affordances of validation and warranting.
+
+3.  Protection from Attack categories
+
+4.  Resilience to Human Error
+
+### Gating Access via Cryptographic Object Capabilities
+
+To use a Holochain application end-users must trigger zome calls that
```

Comment:
This is the first time "zome" has been introduced. The term should be introduced (capitalized) somewhere earlier, probably when introducing DNA. By the way, "DNAs" (as a countable noun) was not really introduced as such.